scripts/tests/test_zkai_deepprove_nanozk_adapter_feasibility.py

```
@@ -0,0 +1,149 @@
+        )
+        self.assertIn("not a DeepProve-1 soundness finding", payload["non_claims"])
+        self.assertIn("not a NANOZK soundness finding", payload["non_claims"])
+
+    def test_validation_rejects_benchmark_overclaims(self) -> None:
+        payload = PROBE.build_probe()
+        payload["systems"][0]["relabeling_benchmark_run"] = True
+        payload["systems_commitment"] = PROBE.blake2b_commitment(
+            payload["systems"], "ptvm:zkai:external-adapter-systems:v1"
+        )
+
+        with self.assertRaisesRegex(PROBE.AdapterFeasibilityError, "benchmark-run overclaim"):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)